californium-core/src/test/java/org/eclipse/californium/core/test/MessageExchangeStoreTool.java

```diff
@@ -186,16 +188,23 @@ public boolean isEmpty() {
 		private final InMemoryObservationStore observationStore;
 		private RequestEventChecker requestChecker;
 
-		private CoapTestEndpoint(InetSocketAddress bind, NetworkConfig config,
+		private CoapTestEndpoint(Connector connector, boolean applyConfiguration, NetworkConfig config,
 				InMemoryObservationStore observationStore, InMemoryMessageExchangeStore exchangeStore,
-				boolean checkAddress) {
-			super(new UDPConnector(bind), true, config, new RandomTokenGenerator(config), observationStore,
-					exchangeStore, new UdpEndpointContextMatcher(checkAddress), COAP_STACK_TEST_FACTORY);
+				EndpointContextMatcher matcher) {
+			super(connector, applyConfiguration, config, new RandomTokenGenerator(config), observationStore,
+					exchangeStore, matcher, COAP_STACK_TEST_FACTORY);
 			this.exchangeStore = exchangeStore;
 			this.observationStore = observationStore;
 			this.requestChecker = new RequestEventChecker();
 		}
 
+		private CoapTestEndpoint(InetSocketAddress bind, NetworkConfig config,
```

Comment:
Maybe you add the stuff to the constructor, which calls this.

Reply:
Sry I didn't get the point.

Reply:
This is an additional private constructor, which is only called by one public constructor. So if the
`new UDPConnector(bind), true, ... new UdpEndpointContextMatcher(checkAddress)` is moved into the public, the private must not be added.

Reply:
Ok I can do that.